### DISCUSSION TITLE
Load component on Taxonomy page

### DISCUSSION THREAD

**Discussion by comsup** [flyntwp/flynt#463] [Q&A] [answered]
Hi there,
I'm loading a component on a taxonomy page and it's work well if there is no item linked to the taxonomy but if there is an item linked the component isn't showing anymore and that's the components of the item who's showing. The idea is to have a Taxonomy page with some components and at the bottom og this page the list of the item of this taxonomy. Is there a "reset_query" or something we have to put somewhere to retrieve the components of the taxonomy after the item loop? I also tried to show the component before the items loop but it's not working. 
Thanks,

**Comment by ben-goodman-uk** [comment]
Is there any movement on this? 

I'm currently experiencing the same issue

**Comment by timohubois** [comment]
Currently, I get not exactly what should be produced exactly. Can someone describe the purpose and the goal more clear for me?

Is the main goal to render a **static** component before or after the “loop” of posts on taxonomy pages? Or should taxonomy pages get a flexible content field like posts, pages to adjust its content? 

@comsup did you found a solution for your purpose?

**Reply by ben-goodman-uk** [reply]
Hi Timo,

For instance, I have a custom taxonomy.php which renders templates/taxonomy.twig, on that taxonomy page I have a grid of posts from a block that I created, similar to the existing archive block, it renders a typical archive page in Flynt.

I wanted to create a taxonomyComponents.php, pretty much a duplicate of page/postComponents.php for adding custom components to the taxonomy and them displaying the components within that taxonomy.twig, I add the Components option and it all displays correctly, in the backend, how you'd expect, you get the component builder on a taxonomy.
  
The fields display perfectly fine in my taxonomy.twig, however, as soon as I add a post to the taxonomy, the post list renders but the taxonomyComponents, no longer renders, if I empty the taxonomy, the taxonomyComponents displays again.

Hope that's clear,

_I have got around this by just rendering the hero block, registering the fields to the taxonomy, and pulling those fields to the taxonomy.php and then rendering the block within the file (by doing some ACF php to get the field content) however it would be nice to offer more flexibility by utilising the actual component builder._

**Comment by comsup** [comment]
Hello Ben,
I don't think it's the best way to achieve this but I retrieved all informations on the functions.php and send them to the twig :

On functions.php :
$query = get_queried_object();  
// getting term datas ---------------------------------------------------------------------------------------
$term = get_term_by('slug', $query->slug, $query->taxonomy);
// geting terms acf datas ------------------------------------------------------------------------------------
$term_acf = get_fields( 'term_'.$term->term_id );
// Adding datas to the return array ---------------------------------------------------------------------------
$data['term'] = $term;
$data['term_acf'] = $term_acf;

// getting datas of the products : ----------------------------------------------------------------
$posts = Timber::get_posts(array('post_type' => 'produits', 'categories_produits' => $term->slug,  'numberposts' => -1));
$arrayProducts = array();
 foreach ($posts as $post):
                  $arrayProducts[$post->conditionnement][] = $post;
endforeach;
$data['sousProduits']  = $arrayProducts;

 return $data;

// And on the twig :
I'm calling my custom H1 of the category with {{ term_acf.h1 }}
and I'm calling the data of the products with a loop : 
      {% for key, produits in sousProduits %}
                    <div class="block-produit">
                        <h3>{{ key }}</h3>
                        {% for key, produit in produits %}
                        <a href="{{ produit.permalink }}" class="lien-produit">{{ produit.name }}</a>
                        {% endfor %}
                    </div>
                {% endfor %}

**Comment by timohubois** [comment]
Thanks for the feedback. The solution from @comsup seems to break out a little bit of timber. 

What do you think about this approach:

1. Register the to the components inside `inc` > `fieldgroups` > `nameOfTheTaxonomyCompononts.php`:
```php
add_action('Flynt/afterRegisterComponents', function () {
    ACFComposer::registerFieldGroup([
        'name' => 'nameOfTheTaxonomyComponents',
        'title' => 'nameOfTheTaxonomy Components',
        'style' => 'seamless',
        'fields' => [
            [
                'name' => 'nameOfTheTaxonomyComponents',
                'label' => __('Page Components', 'flynt'),
                'type' => 'flexible_content',
                'button_label' => __('Add Component', 'flynt'),
                'layouts' => [
                    Components\BlockWysiwyg\getACFLayout(),
                ]
            ]
        ],
        'location' => [
            [
                [
                    'param' => 'taxonomy',
                    'operator' => '==',
                    'value' => 'nameOfTheTaxonomy'
                ]
            ]
        ]
    ]);
});
```

2. Add the term to the timber context inside `taxonomy-nameOfTheTaxonomy.php` file:
```php
use Timber\Timber;
use Timber\Term;

$context = Timber::get_context();
$context['term'] = new Term();

Timber::render('templates/nameOfTheTaxonomy-taxonomy.twig', $context);
```
When you are using the [flnyt next branch](https://github.com/flyntwp/flynt/tree/next), step 2. shouldn’t be necessary, because it uses the Timber [2.0.0-beta.1 ](https://github.com/timber/timber/releases/tag/2.0.0-beta.1) and the [term will added directly to the context](https://github.com/timber/timber/blob/3ac8aca653ebcab5ea524664fab07c761841a7a1/src/Timber.php#L1185).

3. Render the components `templates` > `nameOfTheTaxonomy-taxonomy.twig`:
```twig
{% extends contentOnly ? 'templates/_contentOnly.twig' : 'templates/_document.twig' %}

{% block content %}
  {# In my test meta did not work #}
  {% for component in term.nameOfTheTaxonomyComponents.nameOfTheTaxonomyComponents %}
    {{ renderComponent(component) }}
  {% endfor %}
{% endblock %}
```

 Hope this helps and let me know if it works for you?

**Reply by ben-goodman-uk** [reply]
I'll give this a go, yeah I definitely want to follow the correct development path for features like this. 

Thank you so much, please never stop developing this project, I can't tell you how much it's made me fall in love with WP development again! So simple at every turn. 

I've only just discovered the next branch, some great stuff in there!

Can I use features with a site developed with the existing master branch or is there a simple method of upgrade?

The reusable component would be a game changer, had this request from a client today!

**Reply by timohubois** [reply]
@ben-goodman-uk many thanks for the great feedback! 

The basic idea of flynt is that the theme is a boilerplate and it should only provide starting point for new projects which then can adapted and customised to any design and the respective needs. Further the new release will have some technically changes – as result of these, it will not be possible to update the theme without much effort, unfortunately.

We are looking forward to the release and are curious about any feedback and think that the new approaches will make creating websites with flynt much more modern, easier and faster  🚀

**Reply by ben-goodman-uk** [reply, marked as answer]
That works for me @timohubois,

I did have to make one change however, in the nameOfTheTaxonomy-taxonomy.twig, I removed the second `productCategoryTaxonomyComponents` in the for statement.

However I can confirm we can both render the components and also show the blocks! 🎉 

Additionally I've copied the reusableComponents into my project and that's also working great!

Thank you again!

```
{% extends contentOnly ? 'templates/_contentOnly.twig' : 'templates/_document.twig' %}

{% block content %}

  {% for component in term.productCategoryTaxonomyComponents %}
    {{ renderComponent(component) }}
  {% endfor %}
	{{ renderComponent('GridPostsArchive', { posts: posts }) }}
{% endblock %}
```
